services/src/datasets/storage.rs

```diff
@@ -129,8 +131,28 @@ pub struct CreateDataSet {
     pub definition: DataSetDefinition,
 }
 
-#[allow(clippy::large_enum_variant)]
 #[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct AutoCreateDataSet {
+    pub upload: UploadId,
+    pub dataset_name: String,
+    pub dataset_description: String,
+    pub main_file: String,
+}
+
+impl UserInput for AutoCreateDataSet {
+    fn validate(&self) -> Result<()> {
+        // TODO: more sophisticated input validation
+        ensure!(
+            !self.main_file.contains('/') && !self.main_file.contains(".."),
+            error::InvalidUploadFileName
+        );
+
```

Comment:
you can add that `dataset_name`, as well as `main_file`, shouldn't be empty.
Of course, more sophisticated checks can come later.

Reply:
f302348cdda31a2953801ac4e9dd025a2e559586